### PR TITLE
feat: Making commentary optional for sharePost

### PR DESCRIPTION
**Pull request by TomDowling**
Making the commentary optional on the sharePost as well as trimming the whitespace leading and trailing on the commentary before creating

### CODE REVIEW

File: src/schema/posts.ts [hunk 1]
```diff
@@ -1286,19 +1286,20 @@ export const resolvers: IResolvers<any, Context> = {
...
         sourceId,
-      }: { id: string; commentary: string; sourceId: string },
+      }: { id: string; commentary?: string; sourceId: string },
       ctx,
       info,
     ): Promise<GQLPost> => {
       await ctx.con.getRepository(Post).findOneByOrFail({ id });
       await ensureSourcePermissions(ctx, sourceId, SourcePermissions.Post);
 
+      const strippedCommentary = commentary.trim();
```
Comment (idoshamun) [inline src/schema/posts.ts:1296]: If commentary is null this will break 😭

**Review by idoshamun — CHANGES_REQUESTED** [review]
Let's add a test where we provide empty commentary to see that everything works

**Review by sshanzel — COMMENTED** [review]
Kindly do the same thing on the `submitExternalLink` too as it is also a mutation for sharing a post 🥳

File: src/schema/posts.ts [hunk 2]
```diff
@@ -1250,6 +1250,7 @@ export const resolvers: IResolvers<any, Context> = {
           throw new ValidationError('URL is not valid');
         }
 
+        const strippedCommentary = commentary ? commentary.trim() : '';
```
Comment (sshanzel) [inline src/schema/posts.ts:1253]: Should we probably keep the `null` value if it is empty? WDYT? @idoshamun

**Review by sshanzel — APPROVED** [review]
Looks good to me overall! 🥳 

Just had one non-blocking comment 🚢

File: src/schema/posts.ts (hunk 2 above)
Comment (idoshamun) [reply src/schema/posts.ts:1253]: Yes, let's set it to null

Comment (idoshamun) [reply src/schema/posts.ts:1253]: Maybe it makes more sense to even put this `commentary ? commentary.trim() : null`` inside the `createSharePost` function. This way we have it only in one place

Comment (TomDowling) [reply src/schema/posts.ts]: Good idea, reduce the duplicate code. This change has been done now

Comment (sshanzel) [reply src/schema/posts.ts]: I noticed the change, but we are now missing the `trim` part of the commentary when there is an actual value. Ido's suggestion would probably be the best route to go about it.

`commentary ? commentary.trim() : null`

**Review by idoshamun — APPROVED** [review]

File: src/entity/posts/utils.ts [hunk 3]
```diff
@@ -232,10 +232,15 @@ export const createSharePost = async (
...
+  commentary: string | null,
   visible = true,
 ): Promise<SharePost> => {
-  await validateCommentary(commentary);
+  let strippedCommentary = commentary;
+
+  if (commentary) {
+    strippedCommentary = commentary.trim();
+    await validateCommentary(strippedCommentary);
+  }
```
Comment (sshanzel) [inline src/entity/posts/utils.ts:243]:
```suggestion
  let strippedCommentary = commentary?.trim();

  if (commentary.length) {
    await validateCommentary(strippedCommentary);
  } else {
    strippedCommentary = null;
  }
```

**Review by sshanzel — APPROVED** [review]
Nice work! Just one last thing before merging. We still don't set it to null but have suggested a commit.

**Review by sshanzel — APPROVED** [review]
Thanks for fixing the tests!
